--- origsrc/evas-0.9.9.063/m4/evas_check_loader.m4	2009-09-27 18:42:43.000000000 -0500
+++ src/evas-0.9.9.063/m4/evas_check_loader.m4	2010-02-03 01:03:58.242260400 -0600
@@ -312,8 +312,8 @@ dnl use: EVAS_CHECK_IMAGE_LOADER(loader,
 AC_DEFUN([EVAS_CHECK_IMAGE_LOADER],
 [
 
-m4_pushdef([UP], m4_toupper([[$1]]))
-m4_pushdef([DOWN], m4_tolower([[$1]]))
+m4_pushdef([UP], m4_toupper([$1]))
+m4_pushdef([DOWN], m4_tolower([$1]))
 
 want_loader="$2"
 want_static_loader="no"
